### PR TITLE
[CCA-233][CCA-281] Prepare docs for AWS network cost allocation GA

### PR DESCRIPTION
**Pull request by mpavlov**
### What does this PR do? What is the motivation?
Customers who enable NPM on their hosts will soon automatically get network cost allocation for these hosts in CCM.

Note on agent version and IMDSv2:
Per the NPM team, AWS customers who have upgraded to IMDSv2 can no longer forward accurate NPM metrics via the datadog agent until they upgrade to agent version `7.58`. Network cost allocation in CCM requires NPM metrics so we want to bump the required version.

Note on ccm_network_config flag:
While we also [support ccm_network_config](https://datadoghq.atlassian.net/wiki/spaces/CCM/pages/2595488509/FAQ#I-don%E2%80%99t-need-NPM-monitoring-on-all-of-my-hosts-but-I-do-need-network-cost-allocation-on-all-of-my-hosts.-Is-this-supported%3F) instead of the standard way of configuring NPM, it is preferred to not share this alternative method publicly as it may unintentionally reduce NPM adoption.

### Merge instructions

- [x] Please merge after reviewing

### CODE REVIEW

File: content/en/cloud_cost_management/container_cost_allocation.md [hunk 1]
```diff
@@ -141,12 +141,15 @@ ECS tasks that run on Fargate are already fully allocated [in the CUR][103]. CCM
...
 
+[Network performance monitoring][105] must be enabled on all AWS hosts to allow accurate data transfer cost allocation. If some hosts do not have Network performance monitoring enabled, the data transfer costs for these hosts will not be allocated and may appear as an n/a bucket depending on filter/groupby conditions.
+
 Datadog supports data transfer cost allocation only through the [standard 6 workload resources][104]. If you use custom workload resources their data transfer costs may only be allocated down to the cluster level and not the node/namespace level.
 
 [101]: /containers/kubernetes/tag/?tab=containerizedagent#node-labels-as-tags
 [102]: /containers/kubernetes/tag/?tab=containerizedagent#pod-labels-as-tags
 [103]: https://docs.aws.amazon.com/cur/latest/userguide/what-is-cur.html
 [104]: https://kubernetes.io/docs/concepts/workloads/
+[105]: /network_monitoring/performance/setup
```
Comment (elihschiff) [inline content/en/cloud_cost_management/container_cost_allocation.md:152]: I might have missed it but this doc doesn't mention npm lite anywhere. Should we mention that adding npm is free in our docs specifically. I think that is pretty important information customers would want to know

**Review by elihschiff — APPROVED** [review]

File: content/en/cloud_cost_management/container_cost_allocation.md (hunk 1 above)
Comment (mpavlov) [reply content/en/cloud_cost_management/container_cost_allocation.md:152]: I thought so as well but NPM team was not too comfortable sharing the npm lite setup in the public docs due to potentially causing a reduction in npm revenue. Discussing with their and our PMs and will update here with the outcome.

Comment (mpavlov) [reply content/en/cloud_cost_management/container_cost_allocation.md:152]: Outcome was:
 * CCM can upsell customers to NPM for specific use cases where CCM network cost allocation is not granular enough to identify cost drivers
 * NPM can track number of CCM orgs who are getting networking data without paying for NPM and count them towards indirect NPM revenue
 * After the above is in place, we can publicly document how to set up npm lite

So this PR is no longer blocked on the above, but it's still blocked on the release of agent version 7.58 which was [expected](https://dd.slack.com/archives/CE4PS8MHS/p1729175219142489) to land this week.

Comment (mpavlov) [reply content/en/cloud_cost_management/container_cost_allocation.md:152]: 7.58 [landed](https://github.com/DataDog/datadog-agent/releases/tag/7.58.0) today

**Review by buraizu — APPROVED** [review]

**Pull request merged**